### ISSUE TITLE
Is there a way to do Pytorch's `index_put` ?

### ISSUE THREAD

**Issue by lucblassel** [tracel-ai/burn#2873]
Firstly, thanks for all the hard work on burn, it's a really cool project!  

I am in the process of porting my PyTorch code to burn, and I am running into an issue. In my python code, I create a matrix that allows me to compute pairwise sums of rows in a tensor: 

```python
n=10
X=torch.randn(n,64,64) # Random input

n_pairs=n*(n-1)//2

row_idx = torch.arange(n_pairs).long()
col_indices = torch.triu_indices(n, n, offset=1)

pairmaker = torch.zeros(n_pairs, n)
pairmaker.index_put((row_idx, col_indices[0]), torch.tensor([1]))
pairmaker.index_put((row_idx, col_indices[1]), torch.tensor([1]))

pair_sums = torch.matmul(pairmaker, X) # sums for every pair of rows in X
```

I am not quite sure how to do this with burn. I can get the index tensors with something like: 

```rust
let n = 10;
let n_pairs = n * (n-1) / 2;

let row_idx = Tensor::<Backend,1,Int>::arange(0..npairs,&device)

// This should be equivalent to torch.triu_indices(n,n,offset=1)
let col_indices = Tensor::<Backend,2>::ones([n,n],&device)
    .triu(1)
    .not_equal_elem(0.)
    .argwhere()
    .transpose();

let mut pairmaker = Tensor::<Backend,2>::zeros([n_pairs,n], &device);

// Index put ? 
```

I have tried to use `select_assign` or `slice_assign` but I cannot get it to work. 
I guess I could just loop over the indices and assign 1 to the corresponding element of `pairmaker` but that seems terribly inefficient.  

Am I missing something ? Or is there currently no way to do this with burn ? 

Thanks for your help!

**Comment by laggui** [comment]
There is no direct equivalent to `index_put` in Burn, but you should be able to achieve what you want with other operations.

Given that you're simply assigning all values of the selected indices to 1, perhaps you could make use of `tensor.mask_fill(mask, 1.)`? Just need to construct the equivalent boolean mask for the row/col indices.
